CMS-Enterprise/easi-app
# [EASI-4457] Dataloaders Refactor

# EASI-4457

## Description

- Adds dataloaders for most of the separate db tables associated with TRB and System intake requests. This should speed up the loading of the admin home page for both System Intake and TRB admin views.
    1. TRB Status dataloader
    1. TRB Requester Component dataloader
    1. TRB Request Form dataloader
    1. TRB Advice Letter dataloader
    1. TRB all feedback dataloader
    1. TRB newest feedback dataloader
    1. TRB Attendees dataloader
    1. TRB Documents dataloader
    1. TRB Admin notes dataloader
    1. TRB Request form funding sources dataloader
    1. TRB Request form system intakes dataloader (this is intakes that are attached when filling the form through LCID lookup)
    1. System Intake funding sources dataloader
    1. System Intake notes dataloader
    1. System Intake Actions dataloader
    1. System Intake Gov Req Feedback dataloader
    1. System Intake Documents dataloader
    1. System Intake Business Case dataloader
    1. Business Case Estimated Lifecycle Costs dataloader
- Allows for adding numerous closed intakes/TRB requests in the seed data since this is a big part of the slowdown on the home pages. 

## How to test this change

- There's a variable at the top of the [seed script](https://github.com/CMS-Enterprise/easi-app/blob/feature/EASI-4457/cmd/devdata/main.go#L34) that can be used to generate lots of requests for local testing. Starting the app and seeding with 500 or so closed requests of each type emulates the current prod behavior pretty closely. Load either admin home page to see how long the spinner remains/the query takes to run.
- Automated tests were also updated to work with the dataloader changes.

## PR Author Checklist




## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
